airbyte-cdk/python/airbyte_cdk/utils/print_buffer.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+import sys
+import time
+from io import StringIO
+from threading import RLock
+from types import TracebackType
+from typing import Optional
+
+
+class PrintBuffer:
```

Comment:
Add a comment string here explaining the magic. Less experienced devs might freak out when then stumble at this a year from now.

Reply:
added